napi_js_props.md

```diff
@@ -0,0 +1,225 @@
+# N-API - JavaScript Data Types and Values 
+
+N-API exposes a set of APIs to interrogate and set properties on JavaScript objects.
+Some of these types are documented under [Section 7](https://tc39.github.io/ecma262/#sec-operations-on-objects) 
+of the [ECMAScript Language Specification](https://tc39.github.io/ecma262/).
+
+Fundamentally, all properties in N-API can be represented in one of the following forms:
+- Named: a simple utf8 encoded string
+- Integer-Indexed: an index value represented by `uint32_t`
+- JavaScript Object: a complex JavaScript Object represented by `napi_value`
+
+N-API values are represented by the type `napi_value`.
+Any N-API API that requires a JavaScript object takes in a `napi_value`
+however, it's the caller's responsibility to make sure that the 
+`napi_value` in question is of the JavaScript type expected by the API.
```

Comment:
Does "JavaScript object" in this paragraph refer to a `napi_value` for which `napi_typeof()` returns `napi_object` or does it refer to an entity that is stored as a `v8::Value` in the V8 backend of N-API?